### PR TITLE
Service test restaurant

### PR DESCRIPTION
**Pull request by DorWolfgorAcademy**

### CODE REVIEW

File: src/test/java/com/service/RestaurantServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+class RestaurantServiceTest {
+
+    private final RestaurantRepository restaurantRepository = Mockito.mock(RestaurantRepository.class);
+
+    private static List<Restaurant> restaurantList = new LinkedList<>();
+
+    @BeforeAll
+    public static void setup() {
+        restaurantList = new LinkedList<>();
+        Chef chef1 = new Chef("dor1","chef 1", "url1",0,new Date(2013, Calendar.FEBRUARY,13));
```
Comment (omrip2) [inline src/test/java/com/service/RestaurantServiceTest.java:35]: Please move all the static data to a separate class

File: src/test/java/com/service/RestaurantServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        Restaurant restaurant1 = new Restaurant("restaurant1","1","url",3,0.3,0.0,new Date(2013, Calendar.FEBRUARY,13),chef1);
+        restaurantList.add(restaurant1);
+    }
+
+    @Test
+    void checkTest() {
```
Comment (omrip2) [inline src/test/java/com/service/RestaurantServiceTest.java:41]: Delete

File: src/test/java/com/service/RestaurantServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    }
+
+    @Test
+    @DisplayName("should get all restaurants")
+    void getAll() {
```
Comment (omrip2) [inline src/test/java/com/service/RestaurantServiceTest.java:47]: change the name of this function to be something more self explanatory such as getAllResturants

**Review by omrip2 — CHANGES_REQUESTED** [review]

File: src/main/java/com/controller/HomeController.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+/*
+ TODO
+ unimplemented endpoint
+ need to return homepage objects
+*/
+@RestController
+public class HomeController {
+    @RequestMapping(value = {"/", "/homepage"})
+    public String  getHomePage(){
```
Comment (omrip2) [inline src/main/java/com/controller/HomeController.java:14]: Add todo comment so you want forget it later

**Review by omrip2 — APPROVED** [review]

File: src/test/java/com/service/RestaurantServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        String expectedMessage = "there is a restaurant like this in the system";
+        String actualMessage = exception.getMessage();
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+    @Test
+    void addRestaurant() {
+        RestaurantService service = new RestaurantService(restaurantRepository);
+        Mockito.when(restaurantRepository.existsById(restaurantList.get(0).getName())).thenReturn(false);
+        Mockito.when(restaurantRepository.save(restaurantList.get(0))).thenReturn(restaurantList.get(0));
+        assertEquals(restaurantList.get(0), service.add(restaurantList.get(0)));
```
[the remark this answers is not in the document]
Comment (DorWolfgorAcademy) [reply src/test/java/com/service/RestaurantServiceTest.java:87]: fixed

File: src/test/java/com/service/RestaurantServiceTest.java (hunk 1 above)
Comment (DorWolfgorAcademy) [reply src/test/java/com/service/RestaurantServiceTest.java:35]: fixed

File: src/test/java/com/service/RestaurantServiceTest.java (hunk 2 above)
Comment (DorWolfgorAcademy) [reply src/test/java/com/service/RestaurantServiceTest.java:41]: fixed

File: src/test/java/com/service/RestaurantServiceTest.java (hunk 3 above)
Comment (DorWolfgorAcademy) [reply src/test/java/com/service/RestaurantServiceTest.java:47]: fixed

File: src/main/java/com/controller/HomeController.java (hunk 4 above)
Comment (DorWolfgorAcademy) [reply src/main/java/com/controller/HomeController.java:14]: fixed

**Pull request merged**